webui/tests/cypress/integration/02-loginout.spec.ts

```diff
@@ -1,55 +1,83 @@
 describe('Sign in/out', () => {
 
-  function checkLoggedIn(username: string): void {
-    // Check for the presence/absence of the icons for the user dropdown and
-    // cluster page link in the top bar, which should be present if and only if
-    // the user is logged in.
-    cy.get('#avatar').should('exist');
-    cy.get('#avatar').should('have.text', username.charAt(0).toUpperCase());
-  }
-
-  function checkLoggedOut(): void {
-    cy.get('#avatar').should('not.exist');
-  }
-
-  function ensureLoggedOut(): void {
-    cy.visit('/ui/logout');
-    checkLoggedOut();
-  }
-
-  it('should log in', () => {
+  before(() => {
+    cy.login();
+  });
+
+  const elmTitleSelector = '#det-main-container div.text-2xl';
+
+  it('should be logged in React side', () => {
+    cy.visit('/det/dashboard');
+    cy.checkLoggedIn('determined');
+  });
+
+  it('should be logged in Elm side', () => {
     cy.visit('/ui/experiments');
-    checkLoggedIn('determined');
+    cy.checkLoggedIn('determined');
   });
 
-  it('should log out', () => {
+  it('should be able to log out from Elm', () => {
     // Open the dropdown menu and click the button to log out.
+    cy.login();
+    cy.visit('/ui');
     cy.get('#avatar').click();
-    cy.get('nav a[href="/ui/logout"]').should('have.lengthOf', 1);
-    cy.get('nav a[href="/ui/logout"]').click();
-    checkLoggedOut();
+    cy.get('nav a[href="/det/logout"]').should('have.lengthOf', 1);
+    cy.get('nav a[href="/det/logout"]').click();
+    cy.checkLoggedOut();
   });
 
-  it('should log back in after logging out', () => {
-    // Logging out above should put us on the login page, so enter the login
-    // information directly.
-    ensureLoggedOut();
-    cy.visit('/ui/login');
+  it('should be able to log out from React', () => {
+    // Open the dropdown menu and click the button to log out.
+    cy.login();
+    cy.visit('/det');
+    cy.get('#avatar').click();
+    // TODO add better identifiers to react Link component. make it an anchor tag?
+    cy.get('.ant-dropdown').contains(/sign out/i).should('have.lengthOf', 1);
+    cy.get('.ant-dropdown').contains(/sign out/i).click();
+    cy.checkLoggedOut();
+  });
+
+  it('should be able to log in', () => {
+    cy.logout();
+    cy.visit('/det/login');
     const username = 'determined';
     // We directly set the value to avoid using the less reliable .type() method
     // from Cypress. We also trigger 'input' event to keep it closer to an actual typing
     // behavior this would help functions relying onInput.
-    // cy.get('input#input-username')
+    // cy.get('form input#basic_username')
     //   .invoke('val', username)
+    //   .trigger('change')
     //   .trigger('input');
 
-    cy.get('input#input-username')
-      .type(username, { delay: 50, force: true });
-
-    cy.get('input#input-username')
+    cy.get('input#basic_username')
```

Comment:
if in the `<Form>` in `Authentication.ts` has `name` changed to `name="login"` then this would be `input#login_username`.

kind of unfortunate that antd uses underscores instead of dashes. I didn't see an easy way to switch to dashes in their docs...

Reply:
yeah would have been nicer. the difference between dashes vs _ for me is that many places (at least in vim) dashes break the work whereas _ does not